src/controllers/user.controller.js

```diff
@@ -2,28 +2,83 @@ const userService = require('../services/user.service');
 
 async function getUser(req, res, next) {
   try {
-    res.json({ status: 'success', data: await userService.getAll() });
+    res.json(await userService.getAll());
   } catch (err) {
     console.error('Error while getting user', err.message);
     next(err);
   }
 }
 
-async function addUser(req, res, next) {
+async function register(req, res) {
   if (req.body.username === undefined || req.body.password === undefined) {
     res.status(400);
-    res.json({ status: 'error', data: 'Username or password undefined' });
+    res.json({ message: 'Username or password undefined' });
     return;
   }
   try {
-    res.json({ status: 'success', data: await userService.add(req.body.username, req.body.password) });
+    res.json(await userService.register(req.body.username, req.body.password));
   } catch (err) {
-    console.error('Error while getting user', err.message);
+    if (err.message === 'User allready exists') {
+      res.status(400);
+      res.json({ message: 'User allready exists' });
+      return;
+    }
+
+    res.status(500).json({ message: err.message });
+  }
+}
+
+async function login(req, res) {
+  if (req.body.username === undefined || req.body.password === undefined) {
+    res.status(400);
+    res.json({ message: 'Username or password undefined' });
+    return;
+  }
+  try {
+    res.json(await userService.login(req.body.username, req.body.password));
+  } catch (err) {
+    if (err.message === 'User does not exist' || err.message === 'Invalid password') {
+      res.status(400);
+      res.json({ message: err.message });
+      return;
+    }
+
+    res.status(500).json({ message: err.message });
+  }
+}
+
+async function me(req, res, next) {
+  try {
+    res.json(await userService.me(req.uuid));
```

Comment:
Maybe use a more informative function name like `getCurrentUser`

Reply:
I would not change the name it currently represents the route that is used i think that is more intuitive